src/Compiler/pars.fsy

```diff
@@ -3450,6 +3456,17 @@ headBindingPattern:
   | headBindingPattern AS constrPattern
       { SynPat.As($1, $3, rhs2 parseState 1 3) }
 
+  | headBindingPattern AS recover
```

Comment:
Just for my education - what code would flow to the first of those rules and what would flow to the second one?

Reply:
The most cases inside patterns were able to use the `recover` one. However, cases like the following didn't work well, so I had to add the second, a somewhat artificial, rule:
```fsharp

let _ as

let x = 1
```

The `x` binding wouldn't parse at all without the second rule.

I don't like the approach here, but that's the best thing I've found so far.